public/nextjs_migration/client/js/settings.js

```diff
@@ -0,0 +1,124 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const settingsPartial = document.querySelector("[data-partial='settings']")
+
+function init () {
+  document.body.addEventListener('email-added', handleEvent)
+}
+
+function handleEvent (e) {
+  switch (true) {
+    case e.type === 'email-added':
+      document.querySelector('dialog[data-partial="addEmail"]')
+        .addEventListener('close', () => {
+          window.location.reload()
+        }, { once: true })
+      break
+  }
+}
+
+const settingsAlertOptionsInputs = document.getElementsByClassName('js-settings-alert-options-input')
+if (settingsAlertOptionsInputs?.length) {
+  for (const inputElement of settingsAlertOptionsInputs) {
+    // For some reason the `checked` property gets unset during page load;
+    // this is an ugly workaround to re-check it, in lieu of having converted
+    // the settings page to proper React.
```

Comment:
priase: Solid work-around fix here.

Reply:
:see_no_evil: I felt dirty writing that, but since we should be turning this into proper React later, it felt like the right trade-off for now.